Lesson08_Task01.py

```diff
@@ -0,0 +1,36 @@
+"""
+Lesson08 Task01.
+
+Given the list of integers ( positive , negative, zeros)
+
+Find max multiplication of three values in list
+l1 = [-7, 1, 2, 0, 3, 12, -24, 5, -1, 6]
+
+Input: [-1, 1, 2, 0, 3, 12, 4, 5, -1, 6]
+Output: Max value = "360". Nums are: (12, 5, 6)
+
+Input: [-7, 1, 2, 0, 3, 12, -24, 5, -1, 6]
+Output: Max value = "2016". Nums are: (-7, 12, -24)
+"""
+
+l1 = [-7, 1, 2, 0, 3, 12, -24, 5, -1, 6]
+
+copy_l1 = l1[:]
+
+# check if the length of the list contains at least 3 val
+if len(copy_l1) < 3:
+    print('At least 3 values should be in the input')
+else:
+    max_mult = copy_l1[0] * copy_l1[1] * copy_l1[2]
+    max_nums = (copy_l1[0], copy_l1[1], copy_l1[2])
+
+    # find the maximum multiplication among three values
+    for i in range(len(copy_l1) - 2):
```

Comment:
Straight solution with complexity O(n^3)  that is acceptable for junior level. But you see always such tasks have something hidden that you need to find.  In this task if we sort list
Maximum could be from two options
max1 = el[0] * [el1] * el[-1]    example: [-24, -7, -1, 0, 1, 2, 3, 5, 6, 12]
max2 = el[-3] * el[-2]*el[-1]    example: [-1, -1, 0, 1, 2, 3, 4, 5, 6, 12]

You can check other lists with different numbers - but max patterns will be the same. 
Then you just need to compare max1 vs max2 and choose correct one . Complexity in this case O(1)
Summary if you able to calculate complexity just keep in mind that iterview tasks in 99% could be solved with complexity up to O(n^2).

Reply:
created new file with this solution